SneakPeek/src/com/cyscorpions/dalejulian/sneakpeek/adapters/SneakerCategoryAdapter.java

```diff
@@ -0,0 +1,62 @@
+package com.cyscorpions.dalejulian.sneakpeek.adapters;
+
+import java.util.ArrayList;
+
+import com.cyscorpions.dalejulian.sneakpeek.R;
+import com.cyscorpions.dalejulian.sneakpeek.R.id;
+import com.cyscorpions.dalejulian.sneakpeek.models.SneakerCategory;
+
+import android.app.Activity;
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.TextView;
+
+public class SneakerCategoryAdapter extends ArrayAdapter<SneakerCategory> {
+	private Context mContext;
+	private int layoutResourceId;
+	// SneakerCategory[] data = null;
+	private ArrayList<SneakerCategory> data;
+
+	// view holder
+	static class CategoryHolder {
+		TextView categoryName, categoryDesc;
+	}
+
+	public SneakerCategoryAdapter(Context appContext, int layoutResourceId,
+			ArrayList<SneakerCategory> data) {
+		super(appContext, layoutResourceId, data);
+		this.mContext = appContext;
+		this.layoutResourceId = layoutResourceId;
+		this.data = data;
+	}
+
+	@Override
+	public View getView(int position, View convertView, ViewGroup parent) {
+		View row = convertView;
+		CategoryHolder holder = null;
+
+		if (row == null) {
+			LayoutInflater inflater = ((Activity) mContext).getLayoutInflater();
+			row = inflater.inflate(layoutResourceId, parent, false);
+
+			holder = new CategoryHolder();
+			holder.categoryName = (TextView) row
+					.findViewById(R.id.txtCategoryName);
+			holder.categoryDesc = (TextView) row
+					.findViewById(R.id.txtCategoryDesc);
+
+			row.setTag(holder);
+		} else {
+			holder = (CategoryHolder) row.getTag();
+		}
+
+		SneakerCategory category = data.get(position);
+		holder.categoryName.setText(category.getName());
```

Comment:
You can use the setTag(int key, Object object) method on the row object to attach the category object on the view itself. That way, you can get the exact object used for the contents on the view through getTag(int key, Object object) and use it to get the name in the OnItemClick method in MainActivity.java. The way you are getting the objects right now will have a problem when the views are recycled. 

Regarding the key to be used in the tagging, define the ids in an ids.xml file like so:
<resources>
    <item type="id" value="integer" name="your_id_name"/>
 </resources>